astropy/cosmology/core.py

```diff
@@ -110,6 +110,22 @@ class Cosmology(metaclass=ABCMeta):
 
     meta = MetaData()
 
+    # Unified I/O read and write methods from ``astropy.cosmology.io``
+    # The 'read' method is None in subclasses.
+    read = UnifiedReadWriteMethod(CosmologyRead)
+    write = UnifiedReadWriteMethod(CosmologyWrite)
+
+    @classmethod
+    def __subclasses__(cls, deep=False):
```

Comment:
A registry might work. I'll experiment. Originally, I settled on ``__all_subclasses__``, which I've now combined into ``__subclasses__`` because subclasses could read/write and I wanted to construct a registry of subclasses of *that* subclass (e.g. calling ``FLRW.read`` for a ``FlatLambdaCDM`` cosmology.  Of course this can quite easily also be accomplished with ``{key: cls for key, cls in COSMOLOGY_REGISTRY.items() if issubclass(cls, FLRW)}``. In fact, the registry might be the better way to go for the entry-points I've been experimenting with.

Reply:
Switched to a registry.